### DISCUSSION TITLE
How does Open-Shell replace the Windows Menu?

### DISCUSSION THREAD

**Discussion by eppic** [Open-Shell/Open-Shell-Menu#1166] [Q&A] [answered]
Sorry if this is off-topic, but i wondered...

I tried to replace the default windows start menu with something else for a time now but other applications like "ueli", "keypirinha", ... work better for me than Open-Shell-Menu - to be honest.

However, no other program seems to be able to set the windows key to use as the default way to open the program.
It is possible to use third-party for this (SharpKeys, PowerToys) - But these get rid of any shortcuts with the Win Key. (Win+1, Win+L, ...)
How does Open-Shell achieve this? Does it edit the registry or even overwrite some parts of original Windows Code?

**Comment by blackcrack** [comment]
no, here is a Discussions board.. it's okey for something..
Open shell is not something like Launchy or this what do you have count out..
it's the Startmenu relpacement (Win95, Win7 and so on) with different Skins who it's change out and an Explorerbar addin up to Windows 10 ..
How it make it directly, ask @ge0rdi directly, he is an Programmer who programming here with and at moment manage he it so far.. maybe answer he how it directly works.. at moment can i give you only an Handbook at hand who i have compile it for OpenShell/Classic Shell hand have try to put all so far together what's can maybe give the infos so far..

[Classic+Open-Shell Almanach.odt](https://github.com/Open-Shell/Open-Shell-Menu/files/9740120/Classic%2BOpen-Shell.Almanach.odt)


best
Blacky

**Comment by ge0rdi** [comment, marked as answer]
`Open-Shell` hooks `Progman` window and intercepts `WM_SYSCOMMAND` message with `SC_TASKLIST` parameter.
That one is sent when Start menu was triggered.

You can see hook establishing code [here](https://github.com/Open-Shell/Open-Shell-Menu/blob/master/Src/StartMenu/StartMenuDLL/StartMenuDLL.cpp#L2873).
And actual processing of mentioned message [here](https://github.com/Open-Shell/Open-Shell-Menu/blob/master/Src/StartMenu/StartMenuDLL/StartMenuDLL.cpp#L3280).
